### PR TITLE
PROGRAMMES-5746 - Fetching comments asynchronously

### PR DESCRIPTION
**Pull request by felipeparaujo**
Also moving to use programmes-caching-library

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: config/services.yaml [hunk 1]
```diff
@@ -95,9 +89,12 @@ services:
 
     BBC\ProgrammesMorphLibrary\MorphClient:
         arguments:
-            - '@logger'
             - '@csa_guzzle.client.default'
+            - '@BBC\ProgrammesCachingLibrary\Cache'
+            - '@logger'
             - '%app.morph.endpoint%'
+            - cacheKeyPrefix: 'blogs-frontend.%cosmos_component_release%.morph'
+
```
Comment (mattgreenham) [inline config/services.yaml]: WHITESPACE AHHH

Comment (felipeparaujo) [reply config/services.yaml]: OH NO, I'M SO SO SORRY!

File: src/Service/CommentsService.php [hunk 2]
```diff
@@ -56,29 +57,24 @@ public function queuePostComments(Blog $blog, Post $post): void
         );
     }
 
-    public function getPostComments(Blog $blog, Post $post): ?MorphView
+    public function getByBlogAndPost(Blog $blog, Post $post): ?MorphView
```
Comment (votemike) [inline src/Service/CommentsService.php]: Where is this used?

File: src/Service/CommentsService.php [hunk 3]
```diff
@@ -56,29 +57,24 @@ public function queuePostComments(Blog $blog, Post $post): void
...
+                'apiKey' => $this->apiKey,
+                'mode' => 'embedded',
+                'idctaEnv' => $this->env,
+                'forumId' => $this->getForumId($blog, $post),
+            ],
+            []
+        );
+    }
+
+    public function getPromiseResult(PromiseInterface $promise): ?MorphView
```
Comment (votemike) [inline src/Service/CommentsService.php]: Is this the best name for this? How about `resolve()` or `wait()`? Or is it even worth having this method at all?

Comment (felipeparaujo) [reply src/Service/CommentsService.php]: we need something that waits on the promise to get the result. I don't mind the name, if you think resolve is better, I'll change it

File: src/Service/CommentsService.php (hunk 2 above)
Comment (felipeparaujo) [reply src/Service/CommentsService.php]: Nowhere yet, but it's good to have a synchronous way to make the request for testing and such.

Comment (votemike) [reply src/Service/CommentsService.php]: Is it....?

Comment (felipeparaujo) [reply src/Service/CommentsService.php]: It helps removing one layer of complexity when debugging. In any case, it comes for free from the library, I don't see any advantage in removing this.

File: config/services.yaml [hunk 4]
```diff
@@ -95,9 +89,11 @@ services:
 
     BBC\ProgrammesMorphLibrary\MorphClient:
         arguments:
-            - '@logger'
             - '@csa_guzzle.client.default'
+            - '@BBC\ProgrammesCachingLibrary\Cache'
+            - '@logger'
             - '%app.morph.endpoint%'
+            - cacheKeyPrefix: 'blogs-frontend.%cosmos_component_release%.morph'
```
Comment (BjornTwachtmann) [inline config/services.yaml]: If we're using the cache class we shouldn't need a separate prefix here, should we?

Comment (felipeparaujo) [reply config/services.yaml]: nope, you're right. I fixed this in clifton, but forgot about it here, it seems

**Pull request merged**